test/integration/testRowMode.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2015-2019 Snowflake Computing Inc. All rights reserved.
+ */
+const assert = require('assert');
+const testUtil = require('./testUtil');
+const RowMode = require('./../../lib/constants/row_mode');
+
+
+describe('Test row mode', function () {
+  this.timeout(5000);
+  let connection;
+  const sql = `select *
+               from (select 'a' as key, 1 as foo, 3 as name) as table1
+                        join (select 'a' as key, 2 as foo, 3 as name2) as table2 on table1.key = table2.key
+                        join (select 'a' as key, 3 as foo) as table3 on table1.key = table3.key`;
+
+  const rowModeOptions = [undefined, RowMode.OBJECT, RowMode.ARRAY];
+  const expectedArray = ['a', 1, 3, 'a', 2, 3, 'a', 3];
+  const expectedObject = {KEY_0: 'a', FOO_1: 1, NAME: 3, KEY_3: 'a', FOO_4: 2, NAME2: 3, KEY_6: 'a', FOO_7: 3};
```

Comment:
Why all columns increments with common sequence? As a client I would expect to increment all separately.

Reply:
isn't that breaking change? a developer was using conversion to objects and expect not to handle duplicates by driver - e.g. taking join column which have the same values in two tables? maybe for objects we should have third row mode e.g. 'object_with_duplicated_columns' to add this feature?